### PR TITLE
fix/wrong-appdebug-debug-ip-layout-location

### PR DESCRIPTION
**Pull request by tianhaoz95**
The location of debug_ip_layout used in AppDebug was wrong. This PR contains propagating the HAL API for getting SYSFS path up to XRT device and use that XRT device API to get debug_ip_layout in AppDebug.

### CODE REVIEW

File: src/runtime_src/xrt/device/device.h [hunk 1]
```diff
@@ -810,6 +810,12 @@ class device
     return m_hal->stopTrace(type);
   }
 
+  hal::operations_result<size_t>
```
Comment (stsoe) [inline src/runtime_src/xrt/device/device.h]: This is not an appropriate C++ interface.  Please use const appropriately and use std::array for wrapping fixed length char* arrays.  But I think this function should return a std::string with the path, not size_t.  So signature should be hal_operations<std::string>
getSysfsPath(const std::string& subdev, const std::stirng& entry);

File: src/runtime_src/xrt/device/hal.h [hunk 2]
```diff
@@ -603,6 +603,12 @@ class device
     return operations_result<size_t>();
   }
 
+  virtual operations_result<size_t>
+  getSysfsPath(std::string subdev, std::string entry, char* path, int size)
```
Comment (stsoe) [inline src/runtime_src/xrt/device/hal.h]: Fix per device.h comments

File: src/runtime_src/xrt/device/device.h (hunk 1 above)
Comment (tianhaoz95) [reply src/runtime_src/xrt/device/device.h]: changed to use `hal::operations_result<std::string>` in the new push

File: src/runtime_src/xrt/device/hal.h (hunk 2 above)
Comment (tianhaoz95) [reply src/runtime_src/xrt/device/hal.h]: changed to use `hal::operations_result<std::string>` in the new push

File: src/runtime_src/xrt/device/hal2.h [hunk 3]
```diff
@@ -574,6 +574,21 @@ class device : public xrt::hal::device
       return hal::operations_result<size_t>();
     return m_ops->mStopTrace(m_handle,type);
   }
+
+  virtual hal::operations_result<std::string>
+  getSysfsPath(const std::string subdev, const std::string entry)
```
Comment (stsoe) [inline src/runtime_src/xrt/device/hal2.h]: Please pass by reference

File: src/runtime_src/xrt/device/hal.h [hunk 4]
```diff
@@ -603,6 +603,12 @@ class device
     return operations_result<size_t>();
   }
 
+  virtual operations_result<std::string>
+  getSysfsPath(const std::string subdev, const std::string entry)
```
Comment (stsoe) [inline src/runtime_src/xrt/device/hal.h]: Please pass by reference, e.g. like you do in device.h

File: src/runtime_src/xrt/device/hal2.h (hunk 3 above)
Comment (tianhaoz95) [reply src/runtime_src/xrt/device/hal2.h]: fixed in new push

File: src/runtime_src/xrt/device/hal.h (hunk 4 above)
Comment (tianhaoz95) [reply src/runtime_src/xrt/device/hal.h]: fixed in new push

**Comment by stsoe** [comment]
ok to test

**Pull request merged**